client/state/post-formats/reducer.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import { isValidStateWithSchema } from 'state/utils';
+import * as schema from './schema';
+import {
+	SERIALIZE,
+	DESERIALIZE,
+	POST_FORMATS_RECEIVE,
+	POST_FORMATS_REQUEST,
+	POST_FORMATS_REQUEST_SUCCESS,
+	POST_FORMATS_REQUEST_FAILURE
+} from 'state/action-types';
+
+/**
+ * Returns the updated requests state after an action has been dispatched. The
+ * state maps site ID keys to whether a request for post formats is in progress.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export function requesting( state = {}, action ) {
```

Comment:
Unsure if it was intentionally not used for the `requesting` reducer, but the nice thing is if you use it without a schema, it will return the default initial state for you, so this could be simplified to:

```js
export const requesting = createReducer( {}, {
	[ POST_FORMATS_REQUEST ]: ( state, { siteId } ) => ( { ...state, [ siteId ]: true } ),
	[ POST_FORMATS_REQUEST_SUCCESS ]: ( state, { siteId } ) => ( { ...state, [ siteId ]: false } ),
	[ POST_FORMATS_REQUEST_FAILURE ]: ( state, { siteId } ) => ( { ...state, [ siteId ]: false } )
} );
```